py/desisim/targets.py

```diff
@@ -128,29 +128,39 @@ def sample_objtype(nobj, flavor):
 
 #- multiprocessing needs one arg, not multiple args
 def _wrap_get_targets(args):
-    nspec, flavor, tileid = args
-    return get_targets(nspec, flavor, tileid)
+    nspec, flavor, tileid, randseed = args
+    return get_targets(nspec, flavor, tileid, randseed=randseed)
     
-def get_targets_parallel(nspec, flavor, tileid=None, nproc=None):
+def get_targets_parallel(nspec, flavor, tileid=None, nproc=None, randseed=None):
     import multiprocessing as mp
     if nproc is None:
         nproc = mp.cpu_count() // 2
-    
+
     #- don't bother with parallelism if there aren't many targets
     if nspec < 20:
         log.debug('Not Parallelizing get_targets for only {} targets'.format(nspec))
+        np.random.seed(randseed)
         return get_targets(nspec, flavor, tileid)
     else:
         nproc = min(nproc, nspec//10)        
         log.debug('Parallelizing get_targets using {} cores'.format(nproc))
         args = list()
         n = nspec // nproc
+        if randseed is not None:
+            #- use large multiplier since we don't want thread 2 of randseed=1
+            #- to produce the same results as thread 1 of randseed=2
+            irandseed = randseed*100000
```

Comment:
This seems a little fragile.  This is probably equally fragile, but in desisim.ELG.make_templates() [here](https://github.com/desihub/desisim/blob/master/py/desisim/templates.py#L501) I use the random seed to create a distribution of other random seeds. Could you do something similar here and create another set of seeds, one for each core?